GVFS/GVFS.Common/ProductUpgraderInfo.cs

```diff
@@ -21,19 +21,35 @@ public static string CurrentGVFSVersion()
             return ProcessHelper.GetCurrentProcessVersion();
         }
 
-        public static string GetUpgradesDirectoryPath()
-        {
-            return GVFSPlatform.Instance.GetDataRootForGVFSComponent(UpgradeDirectoryName);
+        public static string GetUpgradeApplicationDirectory()
+        {
+            return Path.Combine(
+                GVFSPlatform.Instance.GetUpgradeProtectedDataDirectory(),
+                ProductUpgraderInfo.ApplicationDirectory);
+        }
+
+        public static string GetParentLogDirectoryPath()
+        {
+            return GVFSPlatform.Instance.GetUpgradeLogDirectoryParentDirectory();
         }
 
         public static string GetLogDirectoryPath()
         {
-            return Path.Combine(GetUpgradesDirectoryPath(), LogDirectory);
+            return Path.Combine(
+                GVFSPlatform.Instance.GetUpgradeLogDirectoryParentDirectory(),
+                ProductUpgraderInfo.LogDirectory);
         }
 
         public static string GetAssetDownloadsPath()
         {
-            return Path.Combine(GetUpgradesDirectoryPath(), DownloadDirectory);
+            return Path.Combine(
+                GVFSPlatform.Instance.GetUpgradeProtectedDataDirectory(),
+                ProductUpgraderInfo.DownloadDirectory);
+        }
+
+        public static string GetHighestAvailableVersionDirectory()

```

Comment:
Can/should this method be removed and replaced with calls to `GVFSPlatform.Instance.GetUpgradeHighestAvailableVersionDirectory()` directly?